src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -0,0 +1,318 @@
+        """
+        This method requests detailed information about an mDNS service, such as its address,
+        port, and other service metadata. The information is requested asynchronously.
+
+        Args:
+            zeroconf (Zeroconf): The Zeroconf instance managing the network operations.
+            service_type (str): The type of the mDNS service to query.
+            name (str): The name of the mDNS service.
+
+        Returns:
+            None: This method does not return any value. The results of the query are processed internally.
+        """
+        # Get service info
+        service_info: AsyncServiceInfo = AsyncServiceInfo(service_type, name)
+        is_service_discovered = await service_info.async_request(zeroconf, 3000)
+
+        if is_service_discovered:
+            # Add service info to discovered services
+            for mdns_type in MdnsType:
+                if service_type == mdns_type.value:
+                    service_info_dict = self.service_info_to_dict(service_info)
+                    self._discovered_services[mdns_type.name].append(service_info_dict)
+                    break
+
+    def service_info_to_dict(self, service_info: AsyncServiceInfo):
+        """
+        Converts an AsyncServiceInfo object to a dictionary for easier access and manipulation.
+
+        This method facilitates the handling of service information by converting the AsyncServiceInfo
+        object, which contains details about an mDNS service, into a more accessible dictionary format.
+
+        Args:
+            service_info (AsyncServiceInfo): The AsyncServiceInfo object to convert.
+
+        Returns:
+            dict: A dictionary representation of the AsyncServiceInfo object.
+        """
+        service_info_dict = {
+            "service_name": service_info.name,
+            "name": service_info.get_name(),
+            "type": service_info.type,
+            "server": service_info.server,
+            "port": service_info.port,
+            "addresses": service_info.parsed_addresses(),
+            "txt_record": service_info.decoded_properties,
+            "priority": service_info.priority,
+            "interface_index": service_info.interface_index,
+            "weight": service_info.weight,
+            "host_ttl": service_info.host_ttl,
+            "other_ttl": service_info.other_ttl
+        }
+        return service_info_dict
+
+    def _get_dut_operational_service_name(self):
+        """
+        Constructs the operational service name for the DUT.
+
+        This method generates a service name string based on the DUT's node ID and the compressed
+        fabric ID, following the operational service naming convention.
+
+        Returns:
+            str: A string representing the operational service name.
+        """
+        node_id = self._tc.dut_node_id
+        node_id_hex = str(hex(int(node_id))[2:].upper()).zfill(16)
```

Comment:
why is the str after upper needed?
could this be:

```python
node_id_hex = f'{node_id:016X}'
```

or

```python
node_id_hex = '%016X' % int(node_id)
```

as that seems shorter?